### PR TITLE
upgraded to xamarin forms, added shell navigation and some lightweigh…

### PR DESCRIPTION
**Pull request by markti**
Beyond just using authentication in a xamarin app I think its important that developers have a clean starting place of how Azure AD B2C is implemented in a modern Xamarin app. This expansion of the demo showcases the following capabilities and practices:

1. Upgraded to Xamarin.Forms 3.6.0.264807
2. Leveraged Xamarin Shell for basic navigation
3. 'auto-login' feature to sign in the user immediately on startup
4. Using MessagingCenter to respond to Authentication events that occur in a central location in multiple places throughout the app (Shell's menu changes based on Log On/Off state)
5. Use of componentized AuthenticationService to act as central repository for authentication state

drawbacks of this demo:
1. UWP is not currently supported by Xamarin Shell so UWP ceases to function.
2. more things than basic sign-on/sign-off...might appear to be over-complicated to a beginning dev

I think these changes should be accepted even with these drawbacks for the following reasons:
1. UWP support will eventually come when the Xamarin Shell team implements Shell for UWP.
2. Most developers looking for Android / iOS productivity anyway. (Sorry MSFT, I still love you, WinPhone 4ever!!!)
3. The most common use of authentication is to do the following:
a) display user information in a visible place in the UI (like on the Shell Menu)
b) allow for user information to be shared across views within the app so you can pass that accessToken from all pages. This implementation is useful even if the user choses not to use Shell. The implementation still applies.
4. the AuthenticationService facade provides sufficient encapsulation that if the dev so chose they could easily rip that out and put it into their own project that wasn't using Shell / MVVM etc.

adding @jennyf19 @bgavrilMS as I originally suggested this update to them...

Log Off command when user is logged in...
image

when logged in we show the user id on the home page...
image

could add a 'user profile page' to display all fields...

works on ios as well...

image

### CODE REVIEW

**Comment by markti** [comment]
added settings to enable / disable forced automatic login 

image

File: UserDetailsClient/UserDetailsClient.iOS/UserDetailsClient.iOS.csproj [hunk 1]
```diff
@@ -108,8 +109,20 @@
     <Reference Include="Microsoft.Identity.Client, Version=3.0.8.0, Culture=neutral, PublicKeyToken=0a613f4dd989e8ae, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Microsoft.Identity.Client.3.0.8\lib\xamarinios10\Microsoft.Identity.Client.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.IdentityModel.JsonWebTokens, Version=5.4.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (bgavrilMS) [inline UserDetailsClient/UserDetailsClient.iOS/UserDetailsClient.iOS.csproj]: Why do you need this? We are trying to discourage people from parsing tokens as: 

- MSAL takes care of all the details around expiration and UPN
- Parsing the token should also involve some cryptographic validation. This is not easy to get right.

**Comment by bgavrilMS** [comment]
Overall looks good, let me know when you are ready and we will review again and merge.

**Comment by charlesroddie** [comment]
I don't think this should be merged. Merging this would make this repo a lot less helpful. It introduces a huge amount of noise into this guide.

> Beyond just using authentication in a xamarin app

active-directory-b2c-xamarin-native provides login and authentication for xamarin apps. There is no need to go beyond. It will just be more distracting and harder to find the relevant information.

> I think its important that developers have a clean starting place of how Azure AD B2C is implemented in a modern Xamarin app.

Agreed and this should be **as clean as possible** and so should be **minimal**. This PR **increases code** and so makes it less of a clean starting point.

> Leveraged Xamarin Shell for basic navigation

Navigation not important to demonstrate here. This repo doesn't need to teach devs how to do navigation.

> Most developers looking for Android / iOS productivity anyway

UWP is a requirement for us.

> The most common use of authentication is to do the following: a) display user information in a visible place in the UI (like on the Shell Menu) b) allow for user information to be shared across views within the app so you can pass that accessToken from all pages.

This repo only needs to show how to get user information. Once the information in a nice .Net class any competent developer can use it anywhere.

This is better as a blog post on Shell and standard C# methods for sharing information rather than as something to put in this repo.

**Comment by jennyf19** [comment]
@markti we've made changes to this sample...what do you want to do with this PR? We could do something like this [sample](https://github.com/Azure-Samples/active-directory-xamarin-native-v2) with an incremental tutorial kind of thing....we can help with that. What do you think?

cc: @jmprieur

**Comment by markti** [comment]
> @markti we've made changes to this sample...what do you want to do with this PR? We could do something like this [sample](https://github.com/Azure-Samples/active-directory-xamarin-native-v2) with an incremental tutorial kind of thing....we can help with that. What do you think?
> 
> cc: @jmprieur

I'm looking at the new changes.

**Pull request closed without merging**